docs/source/cpp/compute.rst

```diff
@@ -1490,6 +1492,9 @@ Structural transforms
   in the list array is appended to the output.  Nulls in the parent list array
   are discarded.
 
+* \(3) For each value in the list child array, the value at index is appended
+  to the output. Nulls in the parent list array are mapped into nulls.
```

Comment:
I think this is clearer. I think it is not necessary to explicitly say anything about nulls here.
```suggestion
* \(3) Output is an array of the same length as the input list array. The
  output values are the values at the specified index of each child list.
```

Reply:
The "append" language made it harder to recognize that this can be a scalar kernel.

Reply:
Thanks Ian!